polyfills/Map/polyfill.js

```diff
@@ -43,11 +43,19 @@
 	var Map = function(data) {
 		this._keys = [];
 		this._values = [];
-
 		// If `data` is iterable (indicated by presence of a forEach method), pre-populate the map
-		data && (typeof data.forEach === 'function') && data.forEach(function (item) {
-			this.set.apply(this, item);
-		}, this);
+		if (data && (typeof data.forEach === 'function')){
+			if (data instanceof Map){
```

Comment:
If you have a Map from an iframe or a worker or something, `Map` is different, and instanceof won't work. What do you think about assuming it is a `Map` if it has all the `Map` instance methods?

```
typeof Map.prototype.clear === 'function' &&
typeof Map.prototype.delete === 'function' &&
typeof Map.prototype.entries === 'function' &&
typeof Map.prototype.forEach === 'function' &&
typeof Map.prototype.get === 'function' &&
typeof Map.prototype.has === 'function' &&
typeof Map.prototype.keys === 'function' &&
typeof Map.prototype.set === 'function' &&
typeof Map.prototype.values === 'function'
```

This is very verbose, happy to discuss alternatives :-)